sycl/test-e2e/AddressSanitizer/common/demangle-kernel-name.cpp

```diff
@@ -0,0 +1,20 @@
+// REQUIRES: linux, cpu, aspect-fp64
+// RUN: %{build} %device_sanitizer_flags -O2 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 %{run} not %t &> %t.txt ; FileCheck --input-file %t.txt %s
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 123456;
+  auto *array = sycl::malloc_device<double>(N, Q);
+
+  Q.submit([&](sycl::handler &h) {
+    h.parallel_for<class MyKernel>(
+        sycl::nd_range<1>(N + 1, 1),
+        [=](sycl::nd_item<1> item) { ++array[item.get_global_id(0)]; });
+  });
```

Comment:
Do we really need to run such a big kernel just to test the output of the sanitizer?

```suggestion
  auto *data = sycl::malloc_device<double>(1, Q);

  Q.single_task<class MyKernel>(
        [=]() { ++data[0]; });
  });
```

Reply:
Using single_task cannot show that the kernel name is demangled as its name is quite a simple one that does not required demangle. So I reduce the range size instead.